Type hinting Added to function and methods

mshemuni opened the pull request:
Type hinting added. see: https://docs.python.org/3/library/typing.html

mshemuni committed:
Type hinting Added to function and methods

davidov541 commented on html_table_parser/parser.py:
```diff
@@ -36,7 +36,7 @@ def __init__(
         self.name = ""
 
-    def handle_starttag(self, tag, attrs):
+    def handle_starttag(self, tag: str, attrs: str) -> None:
```
attrs should be a list, not a str here.

mshemuni replied:
Thank you for the request. I fixed it.

One can ask why I didn't use `list` and imported the `List` from `typing`. The reason is that "deep type hinting" cannot be done with the built-in `list` in `python3.5`.

davidov541 reviewed (changes requested):
Just one place that looks wrong. Otherwise thanks for taking care of this to make it a bit easier to use!

mshemuni committed:
A parameter was wrongly labeled as `str` where it should be a `list`

Thanks to https://github.com/schmijos/html-table-parser-python3/pull/9#pullrequestreview-1161062360 we found a problem and this commit will fix it.

davidov541 commented:
Thank you for your submission! I'm going to merge this now. 

@schmijos - I'm not sure what it takes to get a new version of this available in pip. Can you take care of that or link me to a guide on how to do it for myself?

schmijos commented:
@davidov541 Do you have a Pypi user, I could add for managing releases?

mshemuni commented:
> @davidov541 Do you have a Pypi user, I could add for managing releases?

Looking at the pypi page it looks like someone else uploaded it.
We should find that person.

schmijos commented:
I've got owner access to that package on Pypi. So that should be fine.

davidov541 commented:
@schmijos - My pypi account is @davidov541.

schmijos commented:
@davidov541 your pypi ownership invite is pending now. Cheers and thank you!
